### PR TITLE
Show DLC icon if the effect of the option depends on DLC availability

### LINKED ISSUE #1632: Show info about missing DLC instead of completely hiding options
### Describe your idea

Rewrite part of mod option generator to display info about missing DLC (icon with tooltip?) instead of completely hiding the option. Users may not notice or even report issues about the mod features which might be available only if certain DLC is active. 

As an example I can point `Old Town` Policy tweaks that are currently available only if `After Dark` is enabled. The other example might be an option for forcing [Stronger Road Condition Effects](https://github.com/CitiesSkylinesMods/TMPE/wiki/Gameplay#road-condition-has-a-bigger-impact-on-vehicle-speed)

### Additional nodes

We should recheck if everything works correctly and particular settings are not applied when specific DLCs are not active.

### PR DESCRIPTION
**Pull request by krzychu124**
Resolves #1632

Instead of hiding options if user doesn't have DLC show as read-only. Affects only 4 checkboxes.

image

Build [zip](https://ci.appveyor.com/api/projects/krzychu124/tmpe/artifacts/TMPE.zip?branch=enhancement/dlc-indicator-for-options)

### CODE REVIEW

**Review by chameleon-tbn — APPROVED** [review]
👍 works

File: TLM/TLM/UI/Helpers/DLCRestrictedCheckboxOption.cs [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+                                       Options.PersistTo scope = Options.PersistTo.Savegame) : base(fieldName, scope) {
+        _requiredDLC = requiredDLC;
+        _readOnly = !SteamHelper.IsDLCOwned(_requiredDLC);
+    }
+
+    public override CheckboxOption AddUI(UIHelperBase container) {
+        UIPanel panel = ((UIPanel)((UIHelper)container).self).AddUIComponent<UIPanel>();
+        panel.autoLayout = false;
+        panel.size = new Vector2(720, 22);//default checkbox template size
+        panel.relativePosition = Vector3.zero;
```
Comment (kianzarrin) [inline TLM/TLM/UI/Helpers/DLCRestrictedCheckboxOption.cs:24]: not important but I think the code works the same without these lines.

**Comment by kianzarrin** [comment]
how do you make sure the options are not applied in game when DLC is absent? code path is not even reached?

File: TLM/TLM/UI/Helpers/DLCRestrictedCheckboxOption.cs (hunk 1 above)
Comment (krzychu124) [reply TLM/TLM/UI/Helpers/DLCRestrictedCheckboxOption.cs:24]: It does not. Default size of panel is 600x900 or something like that and is not added as first panel (rel. 0,0)
Notice I added panel and then controls to the panel to be able to shift everything imitating negative left margin.

**Comment by krzychu124** [comment]
> how do you make sure the options are not applied in game when DLC is absent? code path is not even reached?

What do you mean by that?
I think you have missed very important line of code (line 17) which means that missing DLC will make the option disabled. 

https://github.com/CitiesSkylinesMods/TMPE/blob/b93c30add87e2aeac27563a2f06fcc1a4392c62d/TLM/TLM/UI/Helpers/DLCRestrictedCheckboxOption.cs#L13-L18

If you ask what about if user have DLC and disabled it? Does it really a problem? Most of the time, especially with those two DLC savegame won't be loadable without those DLCs and.. user already had to have the DLC to enable the option in the first place, not to mention those settings are ignored when there is no DLC since the code is triggered by DLC content (e.g. evacuation buses may ignore arrows - you needs to have access to Evacuation bus and running disaster which requires DLC).

**Comment by kianzarrin** [comment]
Just because UI is disabled doesn't mean the option won't return a value.

**Review by kianzarrin — APPROVED** [review]

**Pull request merged**